hydra-node/src/Hydra/Chain/Direct.hs

```diff
@@ -122,68 +446,81 @@ mockChainSyncServer queue =
+  serverStIdle :: Int -> ServerStIdle Block (Point Block) (Tip Block) m ()
+  serverStIdle cursor =
     ServerStIdle
       { -- recvMsgRequestNext   :: m (Either (ServerStNext header point tip m a) (m (ServerStNext header point tip m a))),
         recvMsgRequestNext = do
-          tx <- atomically $ readTQueue queue
-          pure $ Left $ SendMsgRollForward (nextBlock tx) tip (mockChainSyncServer queue)
+          tx <- atomically $ do
+            txs <- readTVar db
+            let ix = length txs - cursor
+            if ix < 0 then retry else pure (txs !! ix)
+          let st = ChainSyncServer $ pure $ serverStIdle (cursor + 1)
+          pure $ Left $ SendMsgRollForward (nextBlock tx) tip st
       , recvMsgFindIntersect = \case
-          [] -> pure $ SendMsgIntersectFound origin tip (ChainSyncServer $ pure serverStIdle)
-          h : _ -> pure $ SendMsgIntersectFound h tip (ChainSyncServer $ pure serverStIdle)
+          [] ->
+            let st = ChainSyncServer $ pure $ serverStIdle cursor
+             in pure $ SendMsgIntersectFound origin tip st
+          h : _ ->
+            let st = ChainSyncServer $ pure $ serverStIdle cursor
+             in pure $ SendMsgIntersectFound h tip st
       , recvMsgDoneClient = pure ()
       }
 
-runMockTxSubmissionServer ::
-  (MonadThrow m, MonadST m, MonadSTM m) =>
-  TQueue m (ValidatedTx Era) ->
-  Channel m LByteString ->
-  m ()
-runMockTxSubmissionServer queue channel =
-  void . runPeer nullTracer txSubmissionCodec channel $
-    localTxSubmissionServerPeer $ pure $ mockTxSubmissionServer queue
-
 mockTxSubmissionServer ::
   MonadSTM m =>
-  TQueue m (ValidatedTx Era) ->
+  TVar m [ValidatedTx Era] ->
   LocalTxSubmissionServer (GenTx Block) reject m ()
-mockTxSubmissionServer queue =
+mockTxSubmissionServer db =
   LocalTxSubmissionServer
     { recvMsgSubmitTx = \tx -> do
         case tx of
           GenTxAlonzo genTx ->
-            atomically $ writeTQueue queue (toValidatedTx genTx)
+            atomically $ modifyTVar' db (toValidatedTx genTx :)
           _ ->
             -- FIXME: This should really fail? (i.e. SubmitFail)
             pure ()
-        pure (LocalTxSubmission.SubmitSuccess, mockTxSubmissionServer queue)
+        pure (LocalTxSubmission.SubmitSuccess, mockTxSubmissionServer db)
     , recvMsgDone = ()
     }
  where
   toValidatedTx :: GenTx (ShelleyBlock Era) -> ValidatedTx Era
   toValidatedTx (ShelleyTx _id tx) = tx
 
 --
--- Codecs
+-- Helpers
 --
 
-nodeToClientVLatest :: NodeToClientVersion
-nodeToClientVLatest =
-  fst $ Map.findMax $ supportedNodeToClientVersions proxy
- where
-  proxy = Proxy @(CardanoBlock StandardCrypto)
+-- | This extract __Alonzo__ transactions from a block. If the block wasn't
+-- produced in the Alonzo era, it returns a empty sequence.
+getAlonzoTxs :: Block -> StrictSeq (ValidatedTx Era)
+getAlonzoTxs = \case
+  BlockAlonzo (ShelleyBlock (Ledger.Block _ txsSeq) _) ->
+    txSeqTxns txsSeq
+  _ ->
+    mempty
```

Comment:
This seems local enough to put next to all the other `where` functions up there. I'd argue that `fromLedgerTx` is more important to have on the top-level.